valhalla/jawn/src/lib/producers/SQSProducer.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  SendMessageBatchCommand,
+  SendMessageCommand,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+import { QueuePayload } from "./types";
+import { RequestResponseTopics } from "./types";
+import { MessageProducer } from "./types";
+import {
+  err,
+  ok,
+  PromiseGenericResult,
+  Result,
+} from "../../packages/common/result";
+import {
+  HeliconeScoresMessage,
+  KafkaMessageContents,
+} from "../handlers/HandlerContext";
+
+export class SQSProducer implements MessageProducer {
+  private sqs: SQSClient;
+
+  constructor() {
+    if (
+      !process.env.AWS_REGION ||
+      !process.env.AWS_ACCESS_KEY_ID ||
+      !process.env.AWS_SECRET_ACCESS_KEY ||
+      !process.env.REQUEST_LOGS_QUEUE_URL
+    ) {
+      throw new Error(
+        "Required AWS SQS environment variables are not set, SQSProducer will not be initialized."
+      );
+    }
+
+    this.sqs = new SQSClient({
+      region: process.env.AWS_REGION,
+      credentials: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+      },
+    });
+  }
+
+  async sendMessages({
+    msgs,
+    topic,
+  }: QueuePayload): PromiseGenericResult<string> {
+    let attempts = 0;
+    const maxAttempts = 3;
+    const timeout = 1000;
+
+    while (attempts < maxAttempts) {
+      try {
+        while (msgs.length > 0) {
+          const batchSize = 10;
+          const batches = msgs.slice(0, batchSize);
+          msgs = msgs.slice(batchSize);
+
+          const command = new SendMessageBatchCommand({
+            QueueUrl: this.queueUrl,
```

Comment:
NOTE: I am going to wait until the consuming logic is merged so i can use the same logic there to get the URL